src/api/uuid.rs

```diff
@@ -0,0 +1,121 @@
+use uuid::Uuid;
+
+const BLUETOOTH_BASE_UUID: u128 = 0x00000000_0000_1000_8000_00805f9b34fb;
+const BLUETOOTH_BASE_MASK: u128 = 0x00000000_ffff_ffff_ffff_ffffffffffff;
+const BLUETOOTH_BASE_MASK_16: u128 = 0xffff0000_ffff_ffff_ffff_ffffffffffff;
+
+/// Convert a 32-bit BLE short UUID to a full 128-bit UUID by filling in the standard Bluetooth Base
+/// UUID.
+pub const fn uuid_from_u32(short: u32) -> Uuid {
+    Uuid::from_u128(BLUETOOTH_BASE_UUID | ((short as u128) << 96))
+}
+
+/// Convert a 16-bit BLE short UUID to a full 128-bit UUID by filling in the standard Bluetooth Base
+/// UUID.
+pub const fn uuid_from_u16(short: u16) -> Uuid {
+    uuid_from_u32(short as u32)
+}
```

Comment:
Could these functions be part of the `BleUuid` trait? it feels more natural to have
```rust
use btleplug::api::uuid::BleUuid;
// ...
let my_32bit_uuid = Uuid::from_u32(...);
let my_16bit_uuid = Uuid::from_u16(...);
```
as opposed to
```rust
use btleplug::api::uuid::{uuid_from_u32, uuid_from_u16};
// ...
let my_32bit_uuid = uuid_from_u32(...);
let my_16bit_uuid = uuid_from_u16(...);
```

You would lose the `const fn`, but that is an unstable feature that may become available in the near future

Reply:
I agree that that would be nice, but I think that being a `const fn` is more important. In my experience so far I quite often want to have a bunch of constants up the top of a program like

```rust
const MY_CHARACTERISTIC: Uuid = uuid_from_u16(0x1234);
```

I'll add a TODO to make these part of the trait once that's allowed.